File: Worker/src/Worker.Domain/Services/CalculatorService.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading.Tasks;
using Worker.Domain.Entities;
using Worker.Domain.Interfaces;

namespace Worker.Domain.Services
{
    public class CalculatorService : ICalculatorService
    {
        private readonly ICalculatorRepository _calculatorRepository;
        public CalculatorService(ICalculatorRepository calculatorRepository)
        {
            _calculatorRepository = calculatorRepository;
        }

        public async Task CalculateSum(string id)
        {
            try
            {
                Calculator calculator = await _calculatorRepository.GetById(id);

                if (calculator == null) return;

                // Altera o status para "Processing"
                calculator.StartHandling();
                await _calculatorRepository.Save(id, calculator);

                // Aguarda 2seg
                System.Threading.Thread.Sleep(2000);

                // Soma
                calculator.Sum();
                await _calculatorRepository.Save(id, calculator);
            }
            catch(Exception e)
            {
                throw new Exception(e.Message);
            }

        }
    }
}

File: API/src/domain/services/Calculator/CalculatorService.ts
import { Calculator } from "../../entities/Calculator";
import { Status } from "../../enums/Status";
import { IPublisher } from "../../interfaces/IPublisher";
import ICalculatorRepository from "../../interfaces/repositories/ICalculatorRepository";
import ICalculatorService from "../../interfaces/services/ICalculatorService";
import CalculatorDTO from "./DTOs/CalculatorDTO";
import NewSumDTO from "./DTOs/NewSumDTO";

export class CalculatorService implements ICalculatorService{

    private readonly _calculatorRepository: ICalculatorRepository;
    private readonly _calculatorPublisher: IPublisher;

    constructor(calculatorRepository: ICalculatorRepository, calculatorPublisher: IPublisher){
        this._calculatorRepository = calculatorRepository;
        this._calculatorPublisher = calculatorPublisher;
    }
    
    async getResult(id: string): Promise<CalculatorDTO | null> {
        try {
            const result = await this._calculatorRepository.getById(id);

            if(!result)
                return null;
            
            return {
                id: result.id,
                number1: result.number1,
                number2: result.number2,
                status: result.status,
                result: result.result,
                creationDateTime: result.creationDateTime,
                finishDateTime: result.finishDateTime
            };
        }
        catch (error: any) {
            throw new Error(error.message);
        }
    }

    async createSum(newSumDTO: NewSumDTO): Promise<CalculatorDTO | null> {
        try {
            const calculator = new Calculator({
                number1: newSumDTO.number1,
                number2: newSumDTO.number2,
                status: Status.Pending
            });

            const result = await this._calculatorRepository.save(calculator);

            if(!result)
                return null;

            // Enqueue
            const enqueue = await this._calculatorPublisher.publishInQueue({id: result.id}, "sum");

            if(!enqueue)
                return null;
            
            return {
                id: result.id,
                number1: result.number1,
                number2: result.number2,
                status: result.status,
                result: result.result,
                creationDateTime: result.creationDateTime,
                finishDateTime: result.finishDateTime
            };
        }
        catch (error: any) {
            throw new Error(error.message);
        }
    }
}